samples/COVID/covid-spfx/src/webparts/covidAdmin/components/organisms/ContactTracing.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from "react";
+import { Logger, LogLevel } from "@pnp/logging";
+
+import cloneDeep from "lodash/cloneDeep";
+import isEqual from "lodash/isEqual";
+import find from "lodash/find";
+import forEach from "lodash/forEach";
+import { Dictionary } from "lodash/index";
```

Comment:
Turns out that loading Dictionary from lodash pulls in the entire lodash library.  If possible, it would be good to replace all the lodash references with @microsoft/sp-lodash-subset where possible (replacing forEach with each I believe as well), and replacing Dictionary references with a simple { [key: string]: targetType[]}

Reply:
I think it's just actually importing all of lodash/index but yes, I see your point and the typing will resolve that way so it's a fine adjustment.  I've recently figured out a better way to use lodash with their lodash-es module build. I tested that here and with your suggested change as well as using the es modules (so webpack will deal better) I got the adaptive card build down an additional 5k over just removing the reference to Dictionary.

With regard to the @microsoft/sp-lodash-subset, are you saying it is loaded on the page as an external (like React)?  If that's true than that's fine, but for me to want to get behind that guidance I would want to see more documentation. Lodash doesn't change a ton but, effectively you're tying people to a specific build like with ReactJS and we need to document that a lot more clearly and share more how that's working.

Reply:
Right.  So when I replaced all the references as mentioned above, I shaved about 20-25k off the bundle (it's 124kb now, down from 150 or so I think).  Yes - sp-lodash-subset is a subset of lodash that we support and put on the page (and use ourselves).  It's supported.  Currently we are using lodash-cli to generate it, which I believe is using 4.17.15.  I see that 4.17.21 is the latest (although lodash-cli has a dev-dep on 4.17.15)  Worth looking into finding a way to bump that if necessary.
The bigger hurdle is pnp/sp if I'm honest.  I'm nervous of having 10-20 cards all importing 100k+ of code.  1-2 megs starts to become quite noticable - particularly on mobile devices.  Chatting with Patrick around that.

Reply:
Yeah, I'm a co-maintainer with him for PnPjs... we just agreed that for v3 we will look to find a way to allow even more refinement of the imports.

Reply:
Yeah, I'm a co-maintainer with him for PnPjs... we just agreed that for v3 we will look to find a way to allow even more refinement of the imports.  And yes I'm saying I got it down to 118k with the lodash-es build

Reply:
Yeah, I'm a co-maintainer with him for PnPjs... we just agreed that for v3 we will look to find a way to allow even more refinement of the imports.  And yes I'm saying I got it down to 118k.  Will discuss with Vesa around changing those references as if this is a references sample I don't think we should be just pointing to what is effectively a "hidden" bundle without making sure we've explained and documented it well.

Reply:
:(  https://docs.microsoft.com/en-us/javascript/api/sp-lodash-subset?view=sp-typescript-latest

Yeah, we'll need to get that filled out.  It's public and supported (and even leveraged by the default generated code).  Need to get some docs up though - had no idea it was empty.